src/libraries/Microsoft.Extensions.FileProviders.Physical/src/PollingFileChangeToken.cs

```diff
@@ -38,7 +38,7 @@ public class PollingFileChangeToken : IPollingChangeToken
         /// <param name="fileInfo">The <see cref="System.IO.FileInfo"/> to poll</param>
         public PollingFileChangeToken(FileInfo fileInfo)
         {
-            _fileInfo = fileInfo;
+            _fileInfo = FileSystemInfoHelper.ResolveFileLinkTarget(fileInfo) ?? fileInfo;
```

Comment:
Will this raise new exceptions if the target of the file is not reachable?  Perhaps we can add a test case for that: watching a normal file, watching a non-existent file, watching a symlinked file, watching a symlink with non-existent target.

Reply:
It shouldn't, `Exists` and `LinkTarget != null` should guard for that.
https://github.com/dotnet/runtime/blob/08233bac350b19ab2501deafbb3d6683621fe1bd/src/libraries/Microsoft.Extensions.FileProviders.Physical/src/Internal/FileSystemInfoHelper.cs#L42-L49

However since these APIs are not atomic, it could be that the link gets deleted between LinkTarget and ResolveLinkTarget. I don't know if there's an atomic way to do this.

Reply:
I think it's worse than just not being atomic, since you capture the `FileInfo` for the lifetime of the token, which might be long.

Here's another scenario.  Suppose that the symlink initially points at one target, but then is changed to point at a new target and that new target has a new modified time?  I think that's actually the scenario in this issue.  If you captured the `FileInfo` from the initial target you'd miss that change since you'd be checking the original target.

Reply:
What happens if the symlinks chain goes through an area that you can't read?

```sh
touch bar
mkdir foo
cd foo
ln -s ../bar .
cd ..
ln -s baz foo/bar
sha256sum bar baz
chmod a-x foo
sha256sum bar baz
```

In our world, baz.LinkTarget should resolve (to foo/bar).  Does foo/bar then report Exists is false since it can't be read (the directory is no longer allowed to be traversed into)?

Reply:
> In our world, baz.LinkTarget should resolve (to foo/bar). Does foo/bar then report Exists is false since it can't be read (the directory is no longer allowed to be traversed into)?

@bartonjs Yes, that's what I would expect. Therefore this should print false:
```
var targetInfo = File.ResolveLinkTarget("baz", returnFinaltarget: true);
Console.Write(targetInfo.Exists);
```

Reply:
Thank you for addressing the link target feedback.  Please consider adding test cases for the deleted link target cases, as they seem unique to this addition as it has to handle non-existent link targets, and link targets that get deleted / added.

Reply:
If the link target does not exists, we are returning the `LastWriteTimeUtc` from the link itself but I think that doesn't tell us anything, I will change it to return `DateTime.MinValue` instead. 
Let me know if you think that could be problematic.